include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -2586,6 +2586,36 @@ namespace internal
   }
 
 
+  template <int dim, typename Number>
+  inline void
+  weight_dofs_on_child(const VectorizedArray<Number> *weights,
```

Comment:
Can we rename this into something more generic, e.g. `weight_fe_q_dofs_by_entity`? Also, I wonder if you would mind injecting the dimension `loop_length` via a template argument for the case this is supported, in order to reduce loop overhead?

Reply:
> Also, I wonder if you would mind injecting the dimension loop_length via a template argument for the case this is supported, in order to reduce loop overhead?

I would have thought that inline is enough. But I can do the change.

Reply:
I am not sure, but my experience is that the compiler feels tempted to not inline such functions (with more than >20 obvious instructions) because it sees it fit for multiple template arguments, reducing the machine code size a bit. Now I won't blame any compiler for doing so (not sure if this is the case here, though), because it is impossible for the compiler to know that different degrees will rarely be executed in close temporal proximity, making the code size reduction effective at all.

Reply:
Done!